src/widget/form/settings/avform.cpp

```diff
@@ -116,6 +117,12 @@ void AVForm::showEvent(QShowEvent* event)
     createVideoSurface();
     getVideoDevices();
 
+    // To don't block GUI thread.
+    QtConcurrent::run([this]()
+    {
+        camVideoSurface->setSource(&camera);
```

Comment:
@Diadlo I don't know how `QtConcurrent::run` works, but maybe the function can't exit, when there's still some code inside a `QtConcurrent::run` executing?